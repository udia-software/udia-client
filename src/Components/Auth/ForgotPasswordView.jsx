// @flow
import React from 'react';

import {
  GridLoadingOverlay,
  Button,
  AuthContainer,
  Form,
  FormContent,
  FormField,
  FormFieldErrors,
  AuthFormFieldset,
  Input,
} from '../Styled';

type Props = {
  loading: boolean,
  loadingText?: string,
  email: string,
  errors: string[],
  requestSent: boolean,
  emailErrors: string[],
  handleChangeEmail: Function,
  handleSubmit: Function,
};

const ForgotPasswordView = ({
  loading,
  loadingText,
  email,
  errors,
  requestSent,
  emailErrors,
  handleChangeEmail,
  handleSubmit,
}: Props) => (
  <AuthContainer>
    <GridLoadingOverlay gridAreaName="form" loading={loading} loadingText={loadingText} />
    <h1>Forgot Password</h1>
    <Form onSubmit={handleSubmit}>
      <AuthFormFieldset>
        <legend>Resetting your password, User?</legend>
        <FormFieldErrors errors={errors} />
        <FormContent>
          <FormField error={emailErrors.length > 0} success={requestSent}>
            <label htmlFor="email">
              Email:
              <Input
                type="email"
                id="email"
                placeholder="email@udia.ca"
                onChange={handleChangeEmail}
                autoComplete="off"
                value={email}
              />
            </label>
            {requestSent && 'Password reset email sent!'}
            <FormFieldErrors errors={emailErrors} />
          </FormField>
          <Button color="yellow">Request Password Reset</Button>
        </FormContent>
      </AuthFormFieldset>
    </Form>
  </AuthContainer>
);

ForgotPasswordView.defaultProps = {
  loadingText: 'Loading',
};

export default ForgotPasswordView;
