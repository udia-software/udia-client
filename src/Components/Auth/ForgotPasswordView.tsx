import React, { ChangeEventHandler, FormEventHandler } from "react";
import { Button } from "../Helpers/Button";
import FieldErrors from "../Helpers/FieldErrors";
import FieldSuccesses from "../Helpers/FieldSuccesses";
import GridTemplateLoadingOverlay from "../Helpers/GridTemplateLoadingOverlay";
import {
  FormContainer,
  FormContent,
  FormField,
  FormInput,
  FormLegend,
  SignViewContainer,
  SignViewTitle
} from "./SignViewShared";

interface IProps {
  loading: boolean;
  loadingText?: string;
  email: string;
  errors: string[];
  requestSent: boolean;
  emailErrors: string[];
  handleChangeEmail: ChangeEventHandler<HTMLInputElement>;
  handleSubmit: FormEventHandler<HTMLFormElement>;
}

const ForgotPasswordView = ({
  loading,
  loadingText,
  email,
  errors,
  requestSent,
  emailErrors,
  handleChangeEmail,
  handleSubmit
}: IProps) => (
  <SignViewContainer>
    <GridTemplateLoadingOverlay
      loading={loading}
      loadingText={loadingText}
      gridAreaName="form"
    />
    <SignViewTitle>Forgot Password</SignViewTitle>
    <FormContainer onSubmit={handleSubmit}>
      <fieldset>
        <FormLegend>Resetting your password, User?</FormLegend>
        <FieldErrors errors={errors} />
        <FormContent>
          <FormField error={emailErrors.length > 0} success={requestSent}>
            <label htmlFor="email">
              Email:
              <FormInput
                type="email"
                id="email"
                placeholder="email@udia.ca"
                onChange={handleChangeEmail}
                autoComplete="off"
                value={email}
              />
            </label>
            <FieldSuccesses
              successes={requestSent ? ["Password reset email sent!"] : []}
            />
            <FieldErrors errors={emailErrors} />
          </FormField>
          <Button>Request Password Reset</Button>
        </FormContent>
      </fieldset>
    </FormContainer>
  </SignViewContainer>
);

export default ForgotPasswordView;
