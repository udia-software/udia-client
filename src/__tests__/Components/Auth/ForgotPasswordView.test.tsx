import React from "react";
import renderer from "react-test-renderer";
import ForgotPasswordView from "../../../Components/Auth/ForgotPasswordView";

it("renders ForgotPasswordView correctly", () => {
  const stubHandleMethod = () => null;
  const tree = renderer
    .create(
      <ForgotPasswordView
        loading={false}
        email=""
        errors={[]}
        requestSent={false}
        emailErrors={[]}
        handleChangeEmail={stubHandleMethod}
        handleSubmit={stubHandleMethod}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
