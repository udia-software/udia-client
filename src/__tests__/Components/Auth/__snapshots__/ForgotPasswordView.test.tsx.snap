// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ForgotPasswordView correctly 1`] = `
<div
  className="sc-ifAKCX crLKJc"
>
  <div
    className="sc-bxivhb rwJTG"
  >
    <svg
      className="sc-htpNat ctIEmo"
      height="80px"
      version="1.1"
      viewBox="0 0 90 90"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
      y="0px"
    >
      <polygon
        className={undefined}
        points="38.182,57.753 20.18,47.91 20.18,42.455 38.182,32.652 38.182,39.074 25.625,45.113 38.182,51.379"
      />
      <polygon
        className={undefined}
        points="39.848,62.08 46.351,27.918 50.136,27.918 43.56,62.08"
      />
      <polygon
        className={undefined}
        points="51.8,57.78 51.8,51.4 64.372,45.181 51.8,39.028 51.8,32.696 69.82,42.5 69.82,47.91"
      />
      <path
        className={undefined}
        d="M44.999,86.031L9.465,65.517V24.484L44.999,3.969l35.536,20.516v41.029L44.999,86.031ZM13.07,63.434l31.929,18.434L76.93,63.434V26.566L44.999,8.131L13.07,26.565V63.434L13.07,63.434Z"
      />
    </svg>
    Loading...
  </div>
  <h1
    className="sc-EHOje jAxroT"
  >
    Forgot Password
  </h1>
  <form
    className="sc-bZQynM jkjxgO"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Resetting your password, User?
      </legend>
      <ul
        className="sc-bdVaJa geqhDw"
      />
      <div
        className="sc-htoDjs gXikFF"
      >
        <div
          className="sc-VigVT fqVYBn"
        >
          <label
            htmlFor="email"
          >
            Email:
            <input
              autoComplete="off"
              className="sc-dnqmqq vznFB"
              id="email"
              onChange={[Function]}
              placeholder="email@udia.ca"
              type="email"
              value=""
            />
          </label>
          <ul
            className="sc-bwzfXH cUxdKH"
          />
          <ul
            className="sc-bdVaJa geqhDw"
          />
        </div>
        <button
          className="sc-jTzLTM kbCNdq"
        >
          Request Password Reset
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;
